services/openvpn/client.go

```diff
@@ -194,8 +214,20 @@ func (c *Client) GetConfig() (connection.ConsumerConfig, error) {
 		return nil, errors.Wrap(err, "failed to get consumer public IP")
 	}
 
+	pool := port.NewPool()
+
+	for i := 0; i < 10; i++ {
+		cp, err := pool.Acquire()
+		if err != nil {
+			return nil, err
+		}
+
+		c.ports = append(c.ports, cp.Num())
+	}
+
 	return &ConsumerConfig{
-		IP: publicIP,
+		IP:    publicIP,
+		Ports: c.ports,
```

Comment:
I am almost confident its not a consumer side which has to pick ports. Single provider might have hundreds of consumers therefore change of collision  of ports will be much lower if provider would select ports to ping from.

Reply:
The consumer picks his ports and the provider picks his ports. 
There is no collision, both sides provides available ports.